File: Exercises/Exercise04/java_dgemm/src/benchmark/Writer.java
package benchmark;

import java.io.BufferedWriter;
import java.io.File;
import java.io.FileWriter;
import java.io.IOException;

public class Writer {
    public void write(long time, int matrix_size, String value_type){
        try (BufferedWriter writer = new BufferedWriter(new FileWriter("..\\Results\\results.csv", true))){
            StringBuilder result = new StringBuilder();
            result.append("java");
            result.append(",");
            result.append(value_type);
            result.append(",");
            result.append(matrix_size);
            result.append(",");
            result.append(String.valueOf(time));

            writer.write(result.toString() + "\n");
        } catch (IOException e){
            System.err.println("Error: " + e.getMessage());
        }
    }
}

File: Exercises/Exercise04/Csharp.Dgemm/Csharp.Dgemm/Writer.cs
using System;
using System.IO;
using System.Text;

namespace Csharp.Dgemm
{
    public class Writer
    {
        public void Write(long time, int matrix_size, string value_type)
        {
            try
            {
                using (StreamWriter sw = new StreamWriter("..\\..\\..\\..\\Results\\results.csv", true))
                {
                    StringBuilder result = new StringBuilder();
                    result.Append("csharp");
                    result.Append(",");
                    result.Append(value_type);
                    result.Append(",");
                    result.Append(matrix_size);
                    result.Append(",");
                    result.Append(time);

                    sw.Write(result.ToString() + "\n");
                }
            }
            catch (IOException e)
            {
                Console.WriteLine(e.Message);
            }
        }
    }
}
